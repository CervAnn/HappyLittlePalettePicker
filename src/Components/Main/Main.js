import React from 'react'
import './Main.css'
import ColorsContainer from '../ColorsContainer/ColorsContainer'
import PaletteForm from '../PaletteForm/PaletteForm'
import PaletteContainer from '../PaletteContainer/PaletteContainer'
import Header from '../Header/Header'
import PropTypes from 'prop-types'



const Main = (props) => {
  return (
    <main className="Main">
      <Header clearFolderID={props.clearFolderID}/>
      {!props.folderID &&
      <ColorsContainer
      folders={props.folders} 
      palettes={props.palettes}
      currentPalette={props.currentPalette} 
      toggleLock={props.toggleLock}
      folderID={props.folderID}
      /> 
      }
      {!props.folderID &&
      <PaletteForm
      currentPalette={props.currentPalette} 
      generateRandomColors={props.generateRandomColors}
      folders={props.folders}
      isLoading={props.isLoading}
      addNewPalette={props.addNewPalette} 
      />
      }
       
      {props.folderID && <PaletteContainer 
        palettes={props.palettes}
        currentPalette={props.currentPalette} 
        folderID={props.folderID}
        deleteAPalette={props.deleteAPalette}
      />}
    </main>
  )
}

export default Main;

Main.propTypes = {
  addNewPalette: PropTypes.func,
  clearFolderID: PropTypes.func,
  currentPalette: PropTypes.array,
  deleteAPalette: PropTypes.func,
  folderID: PropTypes.string,
  folders: PropTypes.array,
  generateRandomColors: PropTypes.func,
  isLoading: PropTypes.bool,
  palettes: PropTypes.array,
  toggleLock: PropTypes.func
} 
