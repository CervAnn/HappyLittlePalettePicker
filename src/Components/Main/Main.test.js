import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Main from './Main'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Main', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Main
    addNewPalette={jest.fn()}
    clearFolderID={jest.fn()}
    currentPalette={[{color: "#2C7559", isLocked: false}, {color: "57d576", isLocked: false}, {color: "#a35c95", isLocked: false}, {color: "#461d33", isLocked: false}, {color: "413134", isLocked: false}]}
    deleteAPalette={jest.fn()}
    folderID={""}
    generateRandomColors={jest.fn()}
    folders={{ folders: [
      {created_at: "2019-10-16", id: 16556, name: "Bold", updated_at: "2019-10-17"}, 
      {created_at: "2019-10-16", id: 16557, name: "Shy", updated_at: "2019-10-17"}, 
      {created_at: "2019-10-16", id: 16558, name: "Animals", updated_at: "2019-10-17"}, 
      {created_at: "2019-10-16", id: 16559, name: "Winding Down", updated_at: "2019-10-17"}, 
      {created_at: "2019-10-16", id: 16560, name: "Monster Mash", updated_at: "2019-10-17"}
    ]}}
    isLoading={false}
    palettes={[]}
    toggleLock={jest.fn()}
    />);
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

})