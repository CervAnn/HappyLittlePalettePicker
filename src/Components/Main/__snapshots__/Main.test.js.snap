// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match the snapshot with all data passed in correctly 1`] = `
<main
  className="Main"
>
  <Header
    clearFolderID={[MockFunction]}
  />
  <ColorsContainer
    currentPalette={
      Array [
        Object {
          "color": "#2C7559",
          "isLocked": false,
        },
        Object {
          "color": "57d576",
          "isLocked": false,
        },
        Object {
          "color": "#a35c95",
          "isLocked": false,
        },
        Object {
          "color": "#461d33",
          "isLocked": false,
        },
        Object {
          "color": "413134",
          "isLocked": false,
        },
      ]
    }
    folderID=""
    folders={
      Object {
        "folders": Array [
          Object {
            "created_at": "2019-10-16",
            "id": 16556,
            "name": "Bold",
            "updated_at": "2019-10-17",
          },
          Object {
            "created_at": "2019-10-16",
            "id": 16557,
            "name": "Shy",
            "updated_at": "2019-10-17",
          },
          Object {
            "created_at": "2019-10-16",
            "id": 16558,
            "name": "Animals",
            "updated_at": "2019-10-17",
          },
          Object {
            "created_at": "2019-10-16",
            "id": 16559,
            "name": "Winding Down",
            "updated_at": "2019-10-17",
          },
          Object {
            "created_at": "2019-10-16",
            "id": 16560,
            "name": "Monster Mash",
            "updated_at": "2019-10-17",
          },
        ],
      }
    }
    palettes={Array []}
    toggleLock={[MockFunction]}
  />
  <PaletteForm
    addNewPalette={[MockFunction]}
    currentPalette={
      Array [
        Object {
          "color": "#2C7559",
          "isLocked": false,
        },
        Object {
          "color": "57d576",
          "isLocked": false,
        },
        Object {
          "color": "#a35c95",
          "isLocked": false,
        },
        Object {
          "color": "#461d33",
          "isLocked": false,
        },
        Object {
          "color": "413134",
          "isLocked": false,
        },
      ]
    }
    folders={
      Object {
        "folders": Array [
          Object {
            "created_at": "2019-10-16",
            "id": 16556,
            "name": "Bold",
            "updated_at": "2019-10-17",
          },
          Object {
            "created_at": "2019-10-16",
            "id": 16557,
            "name": "Shy",
            "updated_at": "2019-10-17",
          },
          Object {
            "created_at": "2019-10-16",
            "id": 16558,
            "name": "Animals",
            "updated_at": "2019-10-17",
          },
          Object {
            "created_at": "2019-10-16",
            "id": 16559,
            "name": "Winding Down",
            "updated_at": "2019-10-17",
          },
          Object {
            "created_at": "2019-10-16",
            "id": 16560,
            "name": "Monster Mash",
            "updated_at": "2019-10-17",
          },
        ],
      }
    }
    generateRandomColors={[MockFunction]}
    isLoading={false}
  />
</main>
`;
